### PR TITLE
binlog: use pumps client to write binlog

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
use pumps client to write binlog.

### What is changed and how it works?

use pumps client to write binlog, and tidb will send binlog to pump cluster.

### Check List 

Tests 

 - Unit test

Code changes

most changed in sessionctx/binloginfo.go and tidb-server/main.go

### CODE REVIEW

File: sessionctx/binloginfo/binloginfo.go [hunk 1]
```diff
@@ -14,48 +14,56 @@
...
+// GetPumpsClient gets the pumps client instance.
+func GetPumpsClient() *pClient.PumpsClient {
+	pumpsClientLock.RLock()
+	client := pumpsClient
+	pumpsClientLock.RUnlock()
 	return client
 }
 
+// IsValidePumpsClient returns true if client is not a nil PumpsClient.
+func IsValidePumpsClient(client interface{}) bool {
```
Comment (coocood) [inline sessionctx/binloginfo/binloginfo.go]: IsValidPumpsClient

File: tidb-server/main.go [hunk 2]
```diff
@@ -27,6 +26,8 @@ import (
 
 	"github.com/juju/errors"
 	"github.com/opentracing/opentracing-go"
+	"github.com/pingcap/pd/pd-client"
+	pClient "github.com/pingcap/tidb-tools/tidb-binlog/pump_client"
```
Comment (coocood) [inline tidb-server/main.go]: The package name should not contain uppercase letters.

**Comment by WangXiangUSTC** [comment]
@coocood PTAL again

**Comment by coocood** [comment]
LGTM

**Comment by july2993** [comment]
LGTM

**Pull request merged**